src/app/components/services/tree-node/tree-node.service.js

```diff
@@ -150,10 +150,24 @@ angular.module('transmartBaseUi').factory('TreeNodeService', ['$q', function ($q
             node.loaded = true;
             deferred.resolve(node.loaded);
         }
-
         return deferred.promise;
     };
 
+
+    /**
+     * @memberof TreeNodeService
+     * @param node
+     * @returns boolean
+     */
+    service.isCategoricalLeafNode = function (node) {
+        if(node.type === 'CATEGORICAL_OPTION') {
```

Comment:
Yes, I understand that. But instead of the if-statement, you could directly return the value of the expression, right?